gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -1585,10 +1594,13 @@ public static void CREATE_OR_DROP_RESERVOIR_REGION(String reservoirRegionName,
   }
 
   public static boolean createOrDropReservoirRegion(String reservoirRegionName,
-      String resolvedBaseName, boolean isDrop) {
+      String resolvedBaseName, boolean isDrop, boolean removeSampler) {
     PartitionedRegion existingRegion = Misc.getReservoirRegionForSampleTable(
         reservoirRegionName);
     if (isDrop) {
+      if(removeSampler) {
+        CallbackFactoryProvider.getStoreCallbacks().removeSampler(resolvedBaseName);
+      }
```

Comment:
Will prefer this to be inside `if (existingRegion != null)`check on the line below.